### ISSUE TITLE
SetItemListener allowing duplicate click on item

### ISSUE THREAD

**Issue by angelocavallet** [jonataswalker/ol-contextmenu#150]
https://github.com/jonataswalker/ol-contextmenu/blob/004e1a550c69669db6bb2d0aaa2456ea9fe502fc/src/internal.js#L222

Hello,
I don't know if this is what it's meant to be but when you do more than one click in an item the callback is getting called as many times as you clicked before closeMenu() is called.

You can simulate this on the example given on README
https://jsfiddle.net/jonataswalker/ooxs1w5d/
Just double click on "Add Marker" and you will see that two markers are added instead one.

I wrote a workaround for this (probably not the best and most elegant solution, but it works) as  follows:

```
        setItemListener(li, index) {
            const this_ = this;
            let statusClick = true; //created a local flag to control when call the callback or not
            if (li && typeof this.items[index].callback === 'function') {
                (function (callback) {
                    li.addEventListener(
                        'click',
                        function (evt) {
                            evt.preventDefault();
                            if(statusClick){ //check if flag is true execute callback
                                statusClick = false; //set flag to false to prevent it get executed more than once
                                const obj = {
                                    coordinate: this_.getCoordinateClicked(),
                                    data: this_.items[index].data || null,
                                };
                                this_.closeMenu();
                                callback(obj, this_.map);
                            }
                        },
                        false,
                    );
                })(this.items[index].callback);
            }
        }

```

**Comment by jonataswalker** [comment]
Hi @angelocavallet would handle a PR with your fix?

**Comment by angelocavallet** [comment]
Yeah, it would be awesome!
I've never handle a PR, so I [read about it ](https://help.github.com/articles/creating-a-pull-request/) and created a branch for it (v1.2.1) but I'm not able to push to origin.
I think you need to allow me to do that, right?

**Comment by jonataswalker** [comment]
Actually you didn't read the right article. Please read [this one](https://drupal.gatech.edu/handbook/using-pull-requests-forks) instead.

**Comment by angelocavallet** [comment]
I think it's done.

**Comment by jonataswalker** [comment]
Sorry @angelocavallet I didn't check the fiddle when you posted.

I've updated it fixing the icons URL and can't see what you've reported. Can you confirm it?

https://jsfiddle.net/jonataswalker/ooxs1w5d/

**Comment by angelocavallet** [comment]
No problem @jonataswalker.
On your fiddle, if you fast double click on "Add a marker" item from contextmenu before the menu get closed, then it's action is executed twice, creating two markers.

**Comment by angelocavallet** [comment]
You can notice that it executed twice when you try to remove the marker, because there is two markers, so you remove the first one and you'll see the other behind

**Comment by jonataswalker** [comment]
Fixed by #153
